TeachingRecordSystem/src/TeachingRecordSystem.Core/Dqt/DataverseAdapter.UpdateTeacher.cs

```diff
@@ -430,13 +437,31 @@ public UpdateTeacherHelper(DataverseAdapter dataverseAdapter, UpdateTeacherComma
             }
         }
 
+        public bool ShouldUpdatePII(bool? allowPiiUpdatesFromRegisterfield)
+        {
+            if (allowPiiUpdatesFromRegisterfield == true && (_command.FirstName.HasValue ||
+                _command.MiddleName.HasValue ||
+                _command.LastName.HasValue ||
+                _command.DateOfBirth.HasValue ||
+                _command.EmailAddress.HasValue ||
+                _command.GenderCode.HasValue))
+            {
+                return true;
+            }
+            return false;
+        }
+
         public Contact CreateContactEntity()
         {
+            var firstAndMiddleNames = $"{_command.FirstName.ValueOr("")} {_command.MiddleName.ValueOr("")}".Split(' ', StringSplitOptions.RemoveEmptyEntries | StringSplitOptions.TrimEntries);
```

Comment:
This should be done in the handler rather than here (that's why the Stated* fields are provided separately)